src/Adaptivity/AdaptedTriangulations.jl

```diff
@@ -0,0 +1,316 @@
+  return is_child(get_adapted_model(t1),get_adapted_model(t2))
+end
+
+is_child(t1::Triangulation,t2::AdaptedTriangulation) = false
+
+is_related(t1::AdaptedTriangulation,t2::Triangulation) = is_child(t1,t2)
+is_related(t1::Triangulation,t2::AdaptedTriangulation) = is_child(t2,t1)
+is_related(t1::AdaptedTriangulation,t2::AdaptedTriangulation) = is_child(t1,t2) || is_child(t2,t1)
+
+
+# Wrap Triangulation API
+function Geometry.get_background_model(t::AdaptedTriangulation)
+  get_background_model(t.trian) # === get_model(get_adapted_model(t))
+end
+
+function Geometry.get_grid(t::AdaptedTriangulation)
+  get_grid(t.trian)
+end
+
+function Geometry.get_glue(t::AdaptedTriangulation,::Val{d}) where d
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::AdaptedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return AdaptedTriangulation(v,t.adapted_model)
+end
+
+# Wrap constructors
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::AdaptedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::AdaptedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::AdaptedTriangulation,args...;kwargs...)
+  return AdaptedTriangulation(Triangulation(trian.trian,args...;kwargs...),trian.adapted_model)
+end
+
+function Geometry.BoundaryTriangulation(model::AdaptedDiscreteModel,args...;kwargs...)
+  trian = BoundaryTriangulation(get_model(model),args...;kwargs...)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.SkeletonTriangulation(model::AdaptedDiscreteModel,face_to_mask::AbstractVector{Bool})
+  trian = SkeletonTriangulation(get_model(model),face_to_mask)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.InterfaceTriangulation(model::AdaptedDiscreteModel, cell_to_is_in::Vector{Bool})
+  trian = InterfaceTriangulation(get_model(model),cell_to_is_in)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.InterfaceTriangulation(model::AdaptedDiscreteModel,cell_to_inout::AbstractVector{<:Integer})
+  trian = InterfaceTriangulation(get_model(model),cell_to_inout)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.is_change_possible(strian::AdaptedTriangulation,ttrian::AdaptedTriangulation)
+  # A) Both Triangulations are exactly the same
+  (strian === ttrian) && (return true)
+
+  # B) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian.trian,ttrian.trian)
+  end
+
+  # C) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
```

Comment:
I am not sure the code below C is actually what we want. I think that type equality is not a sufficient/necessary condition to be able to make the change. 

At this point we know that `get_background_model(strian) === get_background_model(ttrian)` is false. 

The first condition that has to be fulfilled to be able to perform the `change` of domain is that 

`is_related(get_background_model(strian),get_background_model(ttrian))` 

is true. This is a necessary, but not sufficient condition. Some extra conditions have to be fulfilled.

Reply:
> Some extra conditions have to be fulfilled.

I think this is harder to define/implement.  I think the extra conditions that have to be fulfilled can be formulated as follows **_in words_** [I do not know yet how to translate the words into code, nor whether we have all the machinery/expressivity required to implement it]:

Let us assume that `strian` is the triangulation corresponding to the parent model, and `ttrian` that of the child model (the reversed case works analogously). Let us introduce an intermediate triangulation `itrian` (we might have to create it? how? might it involve info in `AdaptivityGlue` ?) defined on the parent model that covers the same domain as `ttrian`. Then change is possible if  `is_change_possible(itrian,strian.trian)` is true.

Reply:
Like you mention,  `is_related(strian,ttrian)` does not give the full story since it basically has two possible scenarios that have to be contemplated: 

1) The two Triangulations are of the same "type" (BodyFitted, Skeleton, Boundary, ...) and therefore we can directly do the change using our new routines. 

2) The two Triangulations are of a different "type" and therefore we need to somehow create an intermediate Triangulation `itrian` to act as a bridge. 

Two solutions come to mind: 

A) The first is to try to be able to somehow get the "type of triangulation", create an intermediate Triangulation and use the machinery we have in order to do the change. 

B) The second solution would be to handle this change further down, probably by dispatching the type of the triangulations in our o2n/n2o change domain routines. But this becomes complicated since the number of cases grows exponentially with each new type of triangulation we want to support. 

I don't know how to confront this for now, but I did notice the problem when developing the module.

Reply:
Maybe some extreme dispatching using meta-programing to avoid code replication could be the answer...